rest/adminapitest/admin_api_test.go

```diff
@@ -1433,6 +1433,393 @@ func TestResyncStop(t *testing.T) {
+	require.NoError(t, metadataIDError)
+
+	dbConfig.Bucket = &badName
+	persistedConfig := rest.DatabaseConfig{
+		Version:    version,
+		MetadataID: metadataID,
+		DbConfig:   dbConfig,
+		SGVersion:  base.ProductVersion.String(),
+	}
+	rt.InsertDbConfigToBucket(&persistedConfig, rt.CustomTestBucket.GetName())
+
+	// asser that the config is picked up as invalid config on server context
+	err = rt.WaitForConditionWithOptions(func() bool {
+		invalidDatabases := rt.ServerContext().AllInvalidDatabases()
+		return len(invalidDatabases) == 1
+	}, 200, 1000)
+	require.NoError(t, err)
+
+	// assert that a request to the database fails with correct error message
+	resp := rt.SendAdminRequest(http.MethodGet, "/db1/_config", "")
+	rest.RequireStatus(t, resp, http.StatusNotFound)
+	assert.Contains(t, resp.Body.String(), "Must update database config immediately")
+}
+
+// TestMismatchedBucketNameOnDbConfigUpdate:
+//   - Create a db on the rest tester
+//   - attempt to update the config to change the bucket name on the config to a mismatched bucket name
+//   - assert the request fails
+func TestMismatchedBucketNameOnDbConfigUpdate(t *testing.T) {
+	if base.TestsRequireBootstrapConnection() {
+		t.Skip("Walrus/Rosmar does not support persistent config")
+	}
+	base.RequireNumTestBuckets(t, 2)
+	tb1 := base.GetPersistentTestBucket(t)
+	defer tb1.Close()
+
+	rt := rest.NewRestTester(t, &rest.RestTesterConfig{
+		CustomTestBucket: base.GetPersistentTestBucket(t),
+		PersistentConfig: true,
+		MutateStartupConfig: func(config *rest.StartupConfig) {
+			// configure the interval time to pick up new configs from the bucket to every 1 seconds
+			config.Bootstrap.ConfigUpdateFrequency = base.NewConfigDuration(1000000000)
+		},
+	})
+	defer rt.Close()
+
+	// create db with correct config
+	dbConfig := rt.NewDbConfig()
+	resp := rt.CreateDatabase("db1", dbConfig)
+	rest.RequireStatus(t, resp, http.StatusCreated)
+
+	// wait for db to come online
+	require.NoError(t, rt.WaitForDBOnline())
+	badName := tb1.GetName()
+	dbConfig.Bucket = &badName
+
+	// assert request fails
+	resp = rt.ReplaceDbConfig("db1", dbConfig)
+	rest.RequireStatus(t, resp, http.StatusNotFound)
+}
+
+// TestMultipleBucketWithBadDbConfigScenario1:
+//   - in bucketA and bucketB, write two db configs with bucket name as bucketC
+//   - Start new rest tester and ensure they aren't picked up as valid configs
+func TestMultipleBucketWithBadDbConfigScenario1(t *testing.T) {
+	if base.TestsRequireBootstrapConnection() {
+		t.Skip("Walrus/Rosmar does not support persistent config")
+	}
+	base.RequireNumTestBuckets(t, 3)
+	tb1 := base.GetPersistentTestBucket(t)
+	defer tb1.Close()
+	tb2 := base.GetPersistentTestBucket(t)
+	defer tb2.Close()
+	tb3 := base.GetPersistentTestBucket(t)
+	defer tb3.Close()
+
+	rt1 := rest.NewRestTester(t, &rest.RestTesterConfig{
+		CustomTestBucket: tb1,
+		PersistentConfig: true,
+		MutateStartupConfig: func(config *rest.StartupConfig) {
+			// configure same config groupID
```

Comment:
nit

```suggestion
			// all RestTesters all this test must use the same config ID
```

pull group id into a constant.

You can also set this with `RestTesterConfig.GroupID`. https://github.com/couchbase/sync_gateway/blob/2689cf8e564c46d5cc38edc01264c0f051a8e433/rest/utilities_testing.go#L230